library/public/types.h

```diff
@@ -82,6 +73,133 @@ struct mesh_t
    */
   F3D_EXPORT std::pair<bool, std::string> isValid() const;
 };
+
+/**
+ * Describe a 3D vector.
+ */
+struct F3D_EXPORT vector3_t
+{
+  vector3_t() = default;
+  vector3_t(double x, double y, double z)
+    : Value{ x, y, z }
+  {
+  }
+  vector3_t(const std::vector<double>& vec)
+  {
+    if (vec.size() != 3)
+    {
+      throw std::runtime_error("cannot create a vector3_t");
```

Comment:
you need to create a dedicated `f3d` exception type for this.

Reply:
where to define it, I cannot define it in options.h or options_tool.h, and what would be a good name.

Reply:
In types.h of course. 

`type_creation_exception` maybe ?